sds/src/secondary_validation/polish_nip_checksum.rs

```diff
@@ -0,0 +1,56 @@
+use crate::secondary_validation::Validator;
+
+pub struct PolishNipChecksum;
+
+const POLISH_NIP_LENGTH: usize = 10;
+
+const POLISH_NIP_MULTIPLIERS: &[u32] = &[6, 5, 7, 2, 3, 4, 5, 6, 7];
+
+impl Validator for PolishNipChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if regex_match.len() != POLISH_NIP_LENGTH {
+            return false;
+        }
+
+        // Check if the string contains only allowed characters
+        if !regex_match.chars().all(|c| c.is_ascii_digit()) {
+            return false;
+        }
```

Comment:
We could make this checksum a bit more future proof by removing the separators instead of declaring the checksum as non valid if there is any separator (the regex should check the number of digits in any case.)

This way, filter_map could he used here
```rust
        let mut digits = regex_match.chars().filter_map(|c| c.to_digit(10));
```